src/crypto/crypto_context.h

```diff
@@ -43,13 +44,31 @@ class SecureContext final : public BaseObject {
 
   const SSLCtxPointer& ctx() const { return ctx_; }
 
+  // Utility method that allows for non-default initialization of
+  // the SecureContext.
+  template <typename Func = void(SSLCtxPointer&)>
+  inline void Initialize(Func fn) {
+    fn(*ctx_);
+  }
```

Comment:
This seems like a complicated way of adding a non-const equivalent of `ctx()`, which is probably easier/more straightforward both here in the definition and in the place where it's actually being used.

Reply:
I went with this because I explicitly wanted it to be clear that this was something different but a non-const `ctx()` is fine too,